core/src/main/java/org/jdbi/v3/core/codec/Codec.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.codec;
+
+import java.util.function.Function;
+
+import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.mapper.ColumnMapper;
+
+/**
+ * A Codec provides a convenient way for a bidirectional mapping of an attribute to a database column.
+ * <p>
+ * Groups a {@link ColumnMapper} and an {@link org.jdbi.v3.core.argument.Argument} mapping function for a given type.
+ */
+public interface Codec<T> {
```

Comment:
since a class can implement multiple interfaces and be a ColumnMapper and ArgumentFactory at the same time IIRC, can we skip the getter methods and let Codec be basically an alias for both interfaces? You would `jdbi.registerColumnMapper(codec); jdbi.registerArgumentFactory(codec);` behind the screen then,

Reply:
that is what the codec factory is about. This is about code convenience, not new functionality. We had tons of proliferation (and skew) with ArgumentFactory and ColumnMapper classes and the codecs allow you to implement all the functionality in a single place.